test/cpp/jit/test_gpu_shift.cpp

```diff
@@ -2859,12 +2866,316 @@ TEST(NVFuserTest, FusionConv2DStaticEvenSizedWindow_CUDA) {
+
+  auto options = at::TensorOptions().dtype(at::kFloat).device(at::kCUDA, 0);
+  at::manual_seed(0);
+  at::Tensor t0 = at::randn({numel_x, numel_y}, options);
+  std::vector<IValue> inputs = {t0};
+  auto outputs = fe.runFusion(inputs);
+
+  auto t1 = t0 + 1;
+  auto t2 = shift(t1, {1, -1});
+  auto t3 = shift(t1, {-1, 1});
+  auto t4 = t2 + t3;
+  std::vector<at::indexing::TensorIndex> indices{
+      at::indexing::Slice(1, -1), at::indexing::Slice(1, -1)};
+  t4 = t4.index(indices);
+  auto ref = t4.sum(at::ArrayRef<int64_t>{0, 1});
+
+  testValidate(&fusion, outputs, inputs, {ref}, __LINE__, __FILE__);
+}
+
+// Split and merge
+TEST(NVFuserTest, FusionShiftNoPadding2_CUDA) {
+  Fusion fusion;
+  FusionGuard fg(&fusion);
+
+  auto tv0 = makeSymbolicTensor(2);
+  fusion.addInput(tv0);
+
+  auto tv1 = add(tv0, new Double(1));
+  auto tv2 = shift(tv1, {1, -1}, false);
+  auto tv3 = shift(tv1, {-1, 1}, false);
+  auto tv4 = add(tv2, tv3);
+  auto tv5 = sum(tv4, {0, 1});
+
+  fusion.addOutput(tv5);
+
+  tv1->setMemoryType(MemoryType::Shared);
+
+  tv5->split(0, 4);
+  tv5->split(-1, 8);
+  tv5->reorder({{1, 2}});
+  tv5->merge(-2, -1);
+
+  tv2->computeAt(tv5, -1);
+  tv3->computeAt(tv5, -1);
+
+  tv1->split(0, 4);
+  tv1->split(-1, 8);
+  tv1->reorder({{1, 2}});
+  tv1->merge(-2, -1);
+
+  tv5->axis(-1)->parallelize(ParallelType::TIDx);
+  tv1->axis(-1)->parallelize(ParallelType::TIDx);
+
+  FusionExecutor fe;
+  fe.compileFusion(&fusion);
+
+  int numel_x = 99;
+  int numel_y = 101;
+
+  auto options = at::TensorOptions().dtype(at::kFloat).device(at::kCUDA, 0);
+  at::manual_seed(0);
+  at::Tensor t0 = at::randn({numel_x, numel_y}, options);
+  std::vector<IValue> inputs = {t0};
+  auto outputs = fe.runFusion(inputs);
+
+  auto t1 = t0 + 1;
+  auto t2 = shift(t1, {1, -1});
+  auto t3 = shift(t1, {-1, 1});
+  auto t4 = t2 + t3;
+  std::vector<at::indexing::TensorIndex> indices{
+      at::indexing::Slice(1, -1), at::indexing::Slice(1, -1)};
+  t4 = t4.index(indices);
+  auto ref = t4.sum(at::ArrayRef<int64_t>{0, 1});
+
+  testValidate(&fusion, outputs, inputs, {ref}, __LINE__, __FILE__);
+}
+
+// Split and merge, then welford
+TEST(NVFuserTest, FusionShiftNoPadding3_CUDA) {
+  Fusion fusion;
+  FusionGuard fg(&fusion);
+
+  auto tv0 = makeSymbolicTensor(2);
+  fusion.addInput(tv0);
+
+  auto tv1 = add(tv0, new Double(1));
+  auto tv2 = shift(tv1, {1, -1}, false);
+  auto tv3 = shift(tv1, {-1, 1}, false);
+  auto tv4 = add(tv2, tv3);
+  auto tvs = Welford(tv4, {0, 1});
+  auto tv_avg = tvs.avg;
+  auto tv_M2 = tvs.var_sum;
+  auto tv_N = tvs.n;
+
+  fusion.addOutput(tv_avg);
+  fusion.addOutput(tv_M2);
+  fusion.addOutput(tv_N);
+
+  tv1->setMemoryType(MemoryType::Shared);
+
+  tv_avg->split(0, 4);
+  tv_avg->split(-1, 8);
+  tv_avg->reorder({{1, 2}});
+  tv_avg->merge(-2, -1);
+
+  tv2->computeAt(tv_avg, -1);
+  tv3->computeAt(tv_avg, -1);
+
+  tv1->split(0, 4);
+  tv1->split(-1, 8);
+  tv1->reorder({{1, 2}});
+  tv1->merge(-2, -1);
```

Comment:
Could you change one of these to use transform propagator and parallelize all like? Can be in a follow up PR.

Reply:
#1092